locust/stats.py

```diff
@@ -995,6 +995,10 @@ def _stats_history_data_rows(self, csv_writer, now):
         Note that this method differs from the other methods as it appends time-stamped data to the file, whereas the other methods overwrites the data.
         """
 
+        # Do not try to write rows with all 0s in stats_history.csv when a test is not running
+        if (self.environment.runner.state == "stopped" or self.environment.runner.state == "ready"):
+            return
```

Comment:
I wonder. What if for some reason the runner state is stopped, but for some reason there is still data coming in? Maybe it is safer to not write when there are only zeros?

Reply:
Like if there is a timining issue...

Reply:
I am also beginning to think that the tests are failing because they expect lines to be written to the CSV and that is exactly what this PR is addressing.

Reply:
I dont think so. Looking at for example TestWebUIFullHistory.test_request_stats_full_history_csv() it expects *valid* data in some of those rows that are now missing, so it seems like this really broke something...

Reply:
👍 Tbh, I dont understand that test case. What does it even do, and does it still make sense?

Reply:
Ah, I understand the test now. Can you modify it so that it actually fires a request event? Testing that there are zero entries in the master stats isnt the same as testing that there is one :)